tests/ui/conftest.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+
+"""Plugin for running UI tests."""
+
+import socket
+
+import pytest
+
+from redash import (
+    create_app,
+    redis_connection,
+)
+from redash.models import db
+
+from tests.factories import Factory
+from tests.ui.live_server import LiveServer
+from tests.ui.pages.login import LoginPage
+
+
+@pytest.fixture(name='host')
+def fixture_host():
+    """Return a host name as str."""
+    return 'localhost'
+
+
+@pytest.fixture(name='port')
+def fixture_port():
+    """Return a free port as int."""
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(('', 0))
+    port = s.getsockname()[1]
+
+    # Avoid 'address already in use' errors
+    s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+    s.close()
+    return port
+
+
+@pytest.fixture(name='app')
+def fixture_app(host, port):
+    """Run setup and teardown code for the Flask app."""
+    app = create_app('testing')
+
+    app.config['TESTING'] = True
+    app.config['SERVER_NAME'] = '{host}:{port}'.format(host=host, port=port)
+
+    app_ctx = app.app_context()
+    app_ctx.push()
+
+    db.session.close()
+    db.drop_all()
+    db.create_all()
```

Comment:
Hmm, recreating the tables for every test that uses that fixture seems expensive. Have you done some tests with regard to change of time of the existing tests?